internal/alb/tg/targethealth.go

```diff
@@ -76,12 +80,20 @@ type targetHealthController struct {
 // SyncTargetsForReconciliation starts a go routine for reconciling pod condition statuses for the given targets in the background until they are healthy in the target group
 func (c *targetHealthController) SyncTargetsForReconciliation(ctx context.Context, t *Targets, desiredTargets []*elbv2.TargetDescription) error {
 	conditionType := podConditionTypeForIngressBackend(t.Ingress, t.Backend)
-
 	targetsToReconcile, err := c.filterTargetsNeedingReconciliation(conditionType, t, desiredTargets)
 	if err != nil {
 		return err
 	}
-
+	// use static condition type if no targets to reconcile with the old type
+	useStaticConditionType := len(targetsToReconcile) == 0
```

Comment:
No need to have this in a variable if it's only used once, just put it in the `if` condition directly.

Reply:
Also, I think this might make trouble when mixing the per-ingress backend and static readiness gates (e.g. during a rolling deployment when someone wants to update from the old to the new format) because the way you have it, you will _only_ get pods with the old type or _only_ pods with the new type. Instead, I'd recommend to wrap the whole implementation of `SyncTargetsForReconciliation` into a loop so that it gets executed twice like
```go
for conditionType := range []api.PodConditionType{
    podConditionTypeForIngressBackend(t.Ingress, t.Backend),
    staticConditionType
} {
    // ...
}
```

Reply:
Had tested rolling deployment with change from old to the new format and it worked fine. But makes sense to use a loop. Will make the change.

Reply:
@devkid Pod readiness condition failed to reconcile when i made it a loop. A deployment with single (old) pod readiness condition type failed to reconcile. Maybe because the watch is the same for the 2 condition types and the 2nd condition type in the loop removes the watch from the 1st condition type. Here's the code that didn't work..

```
func (c *targetHealthController) SyncTargetsForReconciliation(ctx context.Context, t *Targets, desiredTargets []*elbv2.TargetDescription) error {
	for _, conditionType := range []api.PodConditionType{
		podConditionTypeForIngressBackend(t.Ingress, t.Backend),
		staticConditionType,
	} {
		targetsToReconcile, err := c.filterTargetsNeedingReconciliation(conditionType, t, desiredTargets)
		if err != nil {
			return err
		}
		// create, update or remove targetGroupWatch for this target group;
		// a targetGroupWatch exists as long as there are targets in the target group whose pod condition statuses need to be reconciled;
		// while the targetGroupWatch exists, a go routine regularly monitors the target health of the targets in the target group and updates the pod condition status for the corresponding pods
		tgWatch, ok := c.tgWatches[t.TgArn]
		if ok {
			if len(targetsToReconcile) == 0 {
				tgWatch.cancel()
				delete(c.tgWatches, t.TgArn)
				return nil
			}
		} else {
			if len(targetsToReconcile) == 0 {
				return nil
			}

			// targetGroupWatch for this target group doesn't exist yet -> create it
			tgWatch, ctx = newTargetGroupWatch(ctx, t.Ingress, t.Backend)
			c.tgWatches[t.TgArn] = tgWatch

			// start watching target health in target group and updating pod condition status
			go c.reconcilePodConditionsLoop(ctx, t.TgArn, conditionType, tgWatch)

		}
		tgWatch.interval <- c.ingressTargetHealthReconciliationInterval(t.Backend.ServiceName, t.Ingress)
		tgWatch.targetsToReconcile <- targetsToReconcile
	}
	return nil
}
```

Reply:
@devkid are you ok with not using a loop or do you have any suggestion on how to use it?

Reply:
I adjusted the PR to work with both readiness :D